src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp

```diff
@@ -741,56 +743,99 @@ void WaveTrackMenuTable::OnMergeStereo(wxCommandEvent &)
    AudacityProject *const project = &mpData->project;
    auto &tracks = TrackList::Get( *project );
 
-   WaveTrack *const pTrack = static_cast<WaveTrack*>(mpData->pTrack);
-   wxASSERT(pTrack);
+   const auto first = tracks.Any<WaveTrack>().find(mpData->pTrack);
+   const auto left = *first;
+   const auto right = *std::next(first);
 
-   auto partner =
-      static_cast<WaveTrack*>(*tracks.Find(pTrack).advance(1));
-
-   if (pTrack->GetRate() != partner->GetRate()) {
-      using namespace BasicUI;
-      ShowMessageBox(XO(
-"Mono tracks must have the same sample rate in order to be combined into a "
-"stereo track"),
-         MessageBoxOptions{}
-           .Caption(XO("Error"))
-           .IconStyle(Icon::Error));
-      return;
-   }
-   if(pTrack->GetSampleFormat() != partner->GetSampleFormat())
+   const auto checkAligned = [](const WaveTrack& left, const WaveTrack& right)
    {
-      BasicUI::ShowMessageBox(XO(
-"Mono tracks must have the same sample format in order to be combined "
-"into a stereo track"),
+      const auto isPlainInterval = [](const WaveTrack::Interval& interval)
+      {
+         return interval.GetTrimLeft() == .0 &&
+            interval.GetTrimRight() == .0 &&
+            interval.GetStretchRatio() == 1.0;
+      };
+      const auto rightIntervals = right.Intervals();
+      for(const auto& a : left.Intervals())
+      {
+         if(!isPlainInterval(*a))
+            return false;
+
+         auto it = std::find_if(
+            rightIntervals.begin(),
+            rightIntervals.end(),
+            [&](const auto& b)
+            {
+               return a->Start() == b->Start() &&
+                  a->End() == b->End() &&
+                  isPlainInterval(*b);
+            });
+         if(it == rightIntervals.end())
+            return false;
+      }
+      return true;
+   };
+
+   if(RealtimeEffectList::Get(*left).GetStatesCount() != 0 ||
+      RealtimeEffectList::Get(*right).GetStatesCount() != 0 ||
+      !checkAligned(*left, *right))
+   {
+      const auto answer = BasicUI::ShowMessageBox(
+         XO(
+"The tracks you are attempting to merge to stereo contain clips at\n"
+"different positions, or otherwise mismatching clips. Merging them\n"
+"will render the tracks.\n\n" 
+"This causes any realtime effects to be applied to the waveform and\n"
+"hidden data to get removed. Additionally, clips may get padded with\n"
```

Comment:
This extra checking and the message are not specified in the Issue.  Did designers specify it?

I would write "be" not "get" twice on this line.  Also, are the embedded newlines really needed, or does the text wrap?

@LWinterberg

Reply:
I put out a message in https://www.figma.com/file/YiOF6yUQS466Fob7bD0vaW/21_WX---Modals?type=design&node-id=2-5&mode=design yes. 

I agree that be is better than get here and I think we can clarify further: 

`"This causes any realtime effects to be applied to the waveform and hidden data to be removed. Additionally, the entire track will become one large clip."`